Firestore/core/include/firebase/firestore/geo_point.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2018 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FIRESTORE_CORE_INCLUDE_FIREBASE_FIRESTORE_GEO_POINT_H_
+#define FIRESTORE_CORE_INCLUDE_FIREBASE_FIRESTORE_GEO_POINT_H_
+
+#include <string.h>
+
+namespace firebase {
+namespace firestore {
+
+/** Immutable class representing a GeoPoint in Firestore */
+class GeoPoint {
+ public:
+  GeoPoint();
```

Comment:
If we intend GeoPoint to be copy/move constructable using the generated constructors we should document it:

```c++
GeoPoint(const GeoPoint& other) = default;
GeoPoint(GeoPoint&& other) = default;
```

This has the extra benefit that should we make a change that breaks the compilers ability to generate these constructors it will fail our build rather than just silently omitting them and breaking our customers' builds.

Reply:
Sorry I should have been clearer in what I meant.

I agree that no compiler would omit these _today_. However, should we accidentally make a change to this class that made it not trivially copyable then we might not realize it before we broke our customers. If you include these declarations and the compiler cannot generate the default implementations because we've made such a change it will fail the build.

Since this is a public API I'm advocating that it's worth it to protect us from ourselves :-).